txscript/opcode.go

```diff
@@ -1043,29 +1068,17 @@ func opcodeCheckLockTimeVerify(op *parsedOpcode, vm *Engine) error {
 		return err
 	}
 
-	// In the rare event that the argument may be < 0 due to some arithmetic
-	// being done first, you can always use 0 OP_MAX OP_CHECKLOCKTIMEVERIFY.
+	// In the rare event that the argument needs to be < 0 due to some
+	// arithmetic being done first, you can always use
+	// 0 OP_MAX OP_CHECKLOCKTIMEVERIFY.
 	if lockTime < 0 {
 		return fmt.Errorf("negative locktime: %d", lockTime)
 	}
 
-	// The lock time field of a transaction is either a block height at
-	// which the transaction is finalized or a timestamp depending on if the
-	// value is before the txscript.LockTimeThreshold.  When it is under the
-	// threshold it is a block height.
-	//
-	// The lockTimes in both the script and transaction must be of the same
-	// type.
-	if !((vm.tx.LockTime < LockTimeThreshold && int64(lockTime) < int64(LockTimeThreshold)) ||
-		(vm.tx.LockTime >= LockTimeThreshold && int64(lockTime) >= int64(LockTimeThreshold))) {
-		return fmt.Errorf("mismatched locktime types -- tx locktime %d, stack "+
-			"locktime %d", vm.tx.LockTime, lockTime)
-	}
-
-	if int64(lockTime) > int64(vm.tx.LockTime) {
-		str := "locktime requirement not satisfied -- locktime is " +
-			"greater than the transaction locktime: %d > %d"
-		return fmt.Errorf(str, lockTime, vm.tx.LockTime)
+	err = verifyLockTime(int64(vm.tx.LockTime), LockTimeThreshold,
```

Comment:
And this will get the rest of the moved comment:

```Go
	// The lock time field of a transaction is either a block height at
	// which the transaction is finalized or a timestamp depending on if the
	// value is before the txscript.LockTimeThreshold.  When it is under the
	// threshold it is a block height.
```